[bug] CMakeConfigDeps seems to generate ghost targets causing build failures

### Describe the bug

Almalinux9 in wsl2, 
conan version 2.18.1 
default profile: (this is not a machine I normally develop on but to get a reproducible example I have set this up fresh on my home machine hence setting config deps in profile against proper protocol)
```
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux

[conf]
tools.cmake.cmakedeps:new=will_break_next
tools.system.package_manager:mode=install
tools.system.package_manager:sudo=True
tools.system.package_manager:tool=dnf 
```

building for example qt/qwt from conancenter recipe fails with:

```
CMake Error at /home/emiliskog/.conan2/p/b/qt9ac7a9c72b494/b/build/Release/generators/libtool-Targets-release.cmake:32 (set_property):
  The link interface of target "libtool::libtool" contains:

    automake::automake

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /home/emiliskog/.conan2/p/b/qt9ac7a9c72b494/b/build/Release/generators/libtoolTargets.cmake:7 (include)
  /home/emiliskog/.conan2/p/b/qt9ac7a9c72b494/b/build/Release/generators/libtool-config.cmake:6 (include)
  /home/emiliskog/.conan2/p/b/cmake1720963180010/p/share/cmake-3.31/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
  /home/emiliskog/.conan2/p/b/qt9ac7a9c72b494/b/build/Release/generators/ODBC-Targets-release.cmake:7 (find_dependency)
  /home/emiliskog/.conan2/p/b/qt9ac7a9c72b494/b/build/Release/generators/ODBCTargets.cmake:7 (include)
  /home/emiliskog/.conan2/p/b/qt9ac7a9c72b494/b/build/Release/generators/ODBCConfig.cmake:6 (include)
  qtbase/cmake/QtFindPackageHelpers.cmake:92 (find_package)
  qtbase/src/plugins/sqldrivers/configure.cmake:20 (qt_find_package)
  qtbase/src/plugins/sqldrivers/CMakeLists.txt:45 (include)

```
 
I assume this happens due to pkg-config files which list no libraries but just executables or config variables but am not entirely sure


### How to reproduce it

with above supplied profile all I did was run this command
`
conan install --requires=qwt/6.3.0 --build=missing
`

--- a/conan/tools/cmake/cmakedeps2/target_configuration.py
+++ b/conan/tools/cmake/cmakedeps2/target_configuration.py
@@ -51,6 +51,8 @@ class TargetConfigurationTemplate2:
         if not requires and not components:  # global cpp_info without components definition
             # require the pkgname::pkgname base (user defined) or INTERFACE base target
             for d in transitive_reqs.values():
+                if d.package_type is PackageType.APP:
+                    continue
                 dep_target = self._cmakedeps.get_property("cmake_target_name", d)
                 dep_target = dep_target or f"{d.ref.name}::{d.ref.name}"
                 link = not (pkg_type is PackageType.SHARED and d.package_type is PackageType.SHARED)

--- a/test/integration/toolchains/cmake/cmakedeps2/test_cmakedeps.py
+++ b/test/integration/toolchains/cmake/cmakedeps2/test_cmakedeps.py
@@ -383,3 +383,102 @@ def test_cmake_find_mode_deprecated():
     args = f"-g CMakeDeps -c tools.cmake.cmakedeps:new={new_value}"
     tc.run(f"install --requires=dep/0.1 {args}")
     assert "CMakeConfigDeps does not support module find mode"
+
+
+def test_requires_to_application():
+    c = TestClient()
+    automake = textwrap.dedent("""
+        from conan import ConanFile
+        class Dependent(ConanFile):
+            name = "automake"
+            version = "0.1"
+            package_type = "application"
+        """)
+
+    conanfile = textwrap.dedent("""
+        from conan import ConanFile
+        class Pkg(ConanFile):
+            name = "libtool"
+            version = "0.1"
+            package_type = "static-library"
+
+            def requirements(self):
+                self.requires('automake/0.1')
+        """)
+    test_package = textwrap.dedent("""
+        from conan import ConanFile
+        from conan.tools.cmake import CMake
+
+        class TestPkg(ConanFile):
+            settings = "os", "compiler", "arch", "build_type"
+            generators = "CMakeDeps", "CMakeToolchain"
+
+            def requirements(self):
+                self.requires(self.tested_reference_str)
+
+            def test(self):
+                pass
+        """)
+
+    c.save({"automake/conanfile.py": automake,
+            "libtool/conanfile.py": conanfile,
+            "libtool/test_package/conanfile.py":test_package})
+    c.run("create automake")
+    c.run(f"create libtool -c tools.cmake.cmakedeps:new={new_value}")
+    targets = c.load("libtool/test_package/libtool-Targets-release.cmake")
+    # The libtool shouldn't depend on the automake::automake target
+    assert "automake::automake" not in targets
+
+
+def test_requires_to_application_component():
+    c = TestClient()
+    automake = textwrap.dedent("""
+        from conan import ConanFile
+        class Dependent(ConanFile):
+            name = "automake"
+            version = "0.1"
+            package_type = "application"
+
+            def package_info(self):
+                self.cpp_info.components["myapp"].exe = "myapp"
+                self.cpp_info.components["myapp"].location = "path/to/myapp"
+                self.cpp_info.components["mylibapp"].type = "header-library"
+        """)
+
+    conanfile = textwrap.dedent("""
+        from conan import ConanFile
+        class Pkg(ConanFile):
+            name = "libtool"
+            version = "0.1"
+            package_type = "static-library"
+
+            def requirements(self):
+                self.requires('automake/0.1')
+            def package_info(self):
+                self.cpp_info.requires = ["automake::mylibapp"]
+        """)
+    test_package = textwrap.dedent("""
+        from conan import ConanFile
+        from conan.tools.cmake import CMake
+
+        class TestPkg(ConanFile):
+            settings = "os", "compiler", "arch", "build_type"
+            generators = "CMakeDeps", "CMakeToolchain"
+
+            def requirements(self):
+                self.requires(self.tested_reference_str)
+
+            def test(self):
+                pass
+        """)
+
+    c.save({"automake/conanfile.py": automake,
+            "libtool/conanfile.py": conanfile,
+            "libtool/test_package/conanfile.py":test_package})
+    c.run("create automake")
+    c.run(f"create libtool -c tools.cmake.cmakedeps:new={new_value}")
+    targets = c.load("libtool/test_package/libtool-Targets-release.cmake")
+    # The libtool shouldn't depend on the automake::automake target
+    assert "automake::automake" not in targets
+    assert "# Requirement automake::mylibapp => Full link: True" in targets
+    assert "$<$<CONFIG:RELEASE>:automake::mylibapp>" in targets

